google/cloud/aiplatform/featurestore/entity_type.py

```diff
@@ -1146,6 +1149,148 @@ def ingest_from_gcs(
             request_metadata=request_metadata,
         )
 
+    def ingest_from_df(
+        self,
+        feature_ids: List[str],
+        feature_time: Union[str, datetime.datetime],
+        df_source: "pd.DataFrame",  # noqa: F821 - skip check for undefined name 'pd'
+        feature_source_fields: Optional[Dict[str, str]] = None,
+        entity_id_field: Optional[str] = None,
+        disable_online_serving: Optional[bool] = None,
+        worker_count: Optional[int] = None,
+        request_metadata: Optional[Sequence[Tuple[str, str]]] = (),
+    ) -> "EntityType":
+        """Ingest feature values from DataFrame.
+
+        Args:
+            feature_ids (List[str]):
+                Required. IDs of the Feature to import values
```

Comment:
nit. Can this be optional? If not set, all columns (except the ID and timestamp column) will be imported

Reply:
Making it optional will lead to breaking changes, assuming we want ingest_from_df, ingest_from_bq, and ingest_from_df has same pattern.

One option is to update `feature_ids: List[str]` to `feature_ids: Union[List[str], str]` and takes `"*"` as value to import all features. 

Though, I think enforcing users to put together a list of `feature_ids` maybe helpful to differentiate created features in the entity_type, versus extra columns in the data source. 

For example, users have some columns does not match to any existing feature_id, and they expect them to be imported. By passing "*" into the service, users will only find out that they need to create the features first after the ingestion. But if they pass in individual feature_id, service will not proceed and throw back features not exist.

Reply:
SG.